built_value/lib/standard_json_plugin.dart

```diff
@@ -59,6 +78,44 @@ class StandardJsonPlugin implements SerializerPlugin {
     return result;
   }
 
+  Map _toMapSpecifyType(List list) {
```

Comment:
_toMapSpecificType ?

Reply:
_toMapWithDiscriminator is a better name. Also went through and added comments to most of the methods in this class, since it's pretty complicated :) thanks.